Libraries/LibCore/TCPServer.cpp

```diff
@@ -32,6 +32,12 @@
 #include <stdio.h>
 #include <sys/socket.h>
 
+#ifndef SOCK_NONBLOCK
+#    include <fcntl.h>
+#    define SOCK_NONBLOCK O_NONBLOCK
+#    define SOCK_CLOEXEC O_CLOEXEC
+#endif
```

Comment:
I don't think this will work. `O_NONBLOCK` and `O_CLOEXEC` are options to `open()`, not to `socket()`.
On macOS, you'll need to call `fcntl()` separately. See for example how it already works in `Core::LocalSocket` :)